From 035b08c32d53298c9d4efe71a1d94d20fbf11ce7 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 6 Jul 2017 08:08:43 -0400
Subject: [PATCH] assertion failed in DestroyImageInfo

A assertion failed in DestroyImageInfo, leading to DOS

bug-debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=870014.
bug: https://github.com/ImageMagick/ImageMagick/issues/547
origin: https://github.com/ImageMagick/ImageMagick/commit/6767f31cac3eacdc9dc41b3193a73bdd37610375

(cherry picked from commit 6767f31cac3eacdc9dc41b3193a73bdd37610375)
---
 coders/mat.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/coders/mat.c b/coders/mat.c
index 4b5edfe75..d160c314e 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -1302,7 +1302,8 @@ done_reading:
   if (quantum_info != (QuantumInfo *) NULL)
     quantum_info=DestroyQuantumInfo(quantum_info);
 END_OF_READING:
-  clone_info=DestroyImageInfo(clone_info);
+  if (clone_info)
+    clone_info=DestroyImageInfo(clone_info);
   CloseBlob(image);
 
 
